client/src/app/components/TargetCard.tsx

```diff
@@ -105,23 +108,23 @@ export const TargetCard: React.FC<TargetCardProps> = ({
     }
   };
 
-  const getImage = (): React.ComponentType => {
-    let result: React.ComponentType<any> = CubesIcon;
-    const imagePath = target?.image?.id
-      ? `/hub/files/${target?.image.id}`
-      : DefaultRulesetIcon;
-    if (target.image) {
-      result = () => (
-        <img
-          src={imagePath}
-          alt="Card logo"
-          style={{ height: 80, pointerEvents: "none" }}
-        />
-      );
+  const fetchImageAndSetDataUrl = async (imagePath: string) => {
+    try {
+      const imageDataResponse = await axios.get(imagePath, {
+        headers: {
+          Accept: "application/octet-stream",
+        },
+      });
+      const encodedSvg = encodeURIComponent(imageDataResponse.data);
+      setImageDataUrl(`data:image/svg+xml,${encodedSvg}`);
```

Comment:
This doesn't work when I create a custom target where the image is a png.

Maybe look at the response's content type (if that is correct at least)?

```js
setImageDataUrl(`data:${imageDataResponse.headers["content-type"]},${encodedSvg}`);
```